dbms/src/Debug/dbgKVStore/dbgFuncInvestigator.cpp

```diff
@@ -0,0 +1,289 @@
+        return;
+    }
+
+    auto table_info = storage->getTableInfo();
+    schema_syncer->syncTableSchema(context, table_info.keyspace_id, table_info.id);
+    if (table_info.partition.num > 0)
+    {
+        for (const auto & def : table_info.partition.definitions)
+        {
+            schema_syncer->syncTableSchema(context, table_info.keyspace_id, def.id);
+        }
+    }
+
+    auto table_id = table_info.id;
+
+    TiKVKey start_key, end_key;
+    HandleID start_handle, end_handle;
+    size_t handle_column_size = table_info.is_common_handle ? table_info.getPrimaryIndexInfo().idx_cols.size() : 1;
+
+    constexpr static size_t OFFSET = 2;
+    size_t arg_size = args.size() - OFFSET;
+    if ((arg_size & 1) != 0)
+    {
+        throw Exception(
+            "Args not matched, should be: database-name, table-name, start1 [, start2, ...], end1, [, end2, ...]",
+            ErrorCodes::BAD_ARGUMENTS);
+    }
+    auto key_size = arg_size / 2;
+
+    if (table_info.is_common_handle)
+    {
+        std::vector<Field> start_field;
+        std::vector<Field> end_field;
+
+        for (size_t i = 0; i < handle_column_size; i++)
+        {
+            auto & column_info = table_info.columns[table_info.getPrimaryIndexInfo().idx_cols[i].offset];
+            TiDB::DatumBumpy start_datum = TiDB::DatumBumpy(
+                RegionBench::convertField(column_info, typeid_cast<const ASTLiteral &>(*args[OFFSET + i]).value),
+                column_info.tp);
+            start_field.emplace_back(start_datum.field());
+            TiDB::DatumBumpy end_datum = TiDB::DatumBumpy(
+                RegionBench::convertField(
+                    column_info,
+                    typeid_cast<const ASTLiteral &>(*args[OFFSET + key_size + i]).value),
+                column_info.tp);
+            end_field.emplace_back(end_datum.field());
+        }
+
+        start_key = RecordKVFormat::genKey(table_info, start_field);
+        end_key = RecordKVFormat::genKey(table_info, end_field);
+    }
+    else
+    {
+        start_handle = static_cast<HandleID>(safeGet<UInt64>(typeid_cast<const ASTLiteral &>(*args[OFFSET]).value));
+        start_key = RecordKVFormat::genKey(table_id, start_handle);
+        end_handle = static_cast<HandleID>(safeGet<UInt64>(typeid_cast<const ASTLiteral &>(*args[OFFSET + 1]).value));
+        end_key = RecordKVFormat::genKey(table_id, end_handle);
+    }
+
+    auto range = RegionRangeKeys(std::move(start_key), std::move(end_key));
+    auto regions = kvstore.getRegionsByRangeOverlap(range.comparableKeys());
+
+    for (const auto & [region_id, region] : regions)
+    {
+        auto r = RegionBench::DebugRegion(region);
+        auto & data = r.debugData();
+
+        for (const auto & [k, v] : data.defaultCF().getData())
+        {
+            if (k.first == start_handle)
+            {
+                result.in_default.emplace_back(std::make_pair(region_id, k.second));
```

Comment:
We should remove `std::make_pair` here which can copy constructor rather than construct in place.

Reply:
fixed